fix: check if tsuru.yaml has process commands before fallbacking to i…

ravilock opened the pull request:
…mage command

ravilock committed:
fix: check if tsuru.yaml has process commands before fallbacking to image command

ravilock committed:
test: add testdata for build with tsuru.yaml with processes

ravilock committed:
test: adjust test assertion

fbtravi commented:
Hi @ravilock 

This change covers the case where we have a tsuru.yaml file defined with processes, and the process does not have a start command.

```shell
processes:
  - name: web
    # Using the command from the Dockerfile
    command: npm start
    healthcheck:
      path: /health
      scheme: http
      allowed_failures: 3
      force_restart: true
      interval_seconds: 60
      timeout_seconds: 60
      deploy_timeout_seconds: 600
```

wpjunior commented on pkg/build/buildkit/testdata/python-tsuru-yaml/tsuru.yaml:
```diff
@@ -0,0 +1,14 @@
+      && echo "DATABASE_PASSWORD=${DATABASE_PASSWORD}" >> /tmp/tsuru/envs
+    - python --version
+
+processes:
```
@ravilock we can test multi processes?

infezek replied:
@ravilock can we test the response without processes?

ravilock committed:
refactor(build): remove tsuru dep, improve local dev

- Replaced usage of github.com/tsuru/tsuru types with local TsuruYamlData
  and related structs in pkg/build/types.go.
- Refactored buildkit build logic to use new tsuru.yaml parsing and
  process handling.
- Updated tests and testdata to support multiple processes and new
  tsuru.yaml structure.
- Added misc/local-dev.sh script for local loopback IP setup/cleanup.
- Improved Makefile with setup/cleanup targets and Docker Compose
  detection.
- Removed tsuru dependency from go.mod/go.sum and added new indirect
  dependencies.

crgisch reviewed (approved):
LGTM 👍 

Would be nice to add on README how to run/test locally using new Makefile commands :)

ravilock committed:
chore: add instructions on how to run project with new script and makefile
